src/backend/mpi/swdsm.cpp

```diff
@@ -1364,29 +1380,36 @@ void storepageDIFF(unsigned long index, unsigned long addr){
 
 	/* CSPext: Pointer to actually used MPI windows */
 	MPI_Win real_globalDataWindow = globalDataWindow[homenode];
-	MPI_Win real_replDataWindow = replDataWindow[repl_node];
-	argo::node_id_t real_home_id = node_alter_tbl[homenode].alter_home_id;
-	argo::node_id_t real_repl_id = node_alter_tbl[repl_node].alter_repl_id;
+	MPI_Win real_replDataWindow = real_globalDataWindow;
+	argo::node_id_t real_home_id = homenode;
+	argo::node_id_t real_repl_id = homenode;
+
 	/* CSP:
-	 * For any operation on a window initialized by MPI_Win_create_dynamic(),
-	 *  always set the target_disp as the true virtual addr on the origin node.
-	 * */
+	* For any operation on a window initialized by MPI_Win_create_dynamic(),
```

Comment:
I think these spaces are needed in this format of comment. They are automatically added by my editor